java/client/src/main/java/glide/api/commands/servermodules/Json.java

```diff
@@ -394,6 +396,244 @@ public static CompletableFuture<GlideString> get(
+     * @param path The path within the JSON document.
+     * @param index The array index before which values are inserted.
+     * @param values The JSON values to be inserted into the array, in JSON formatted bytes or str.
+     *     JSON string values must be wrapped with quotes. For example, to append <code>"foo"</code>,
+     *     pass <code>"\"foo\""</code>.
+     * @return
+     *     <ul>
+     *       <li>For JSONPath (<code>path</code> starts with <code>$</code>):<br>
+     *           Returns an <code>Object[]</code> with a list of integers for every possible path,
+     *           indicating the new length of the array, or <code>null</code> for JSON values matching
+     *           the path that are not an array. If <code>path</code> does not exist, an empty array
+     *           will be returned.
+     *       <li>For legacy path (<code>path</code> doesn't start with <code>$</code>):<br>
+     *           Returns an integer representing the new length of the array. If multiple paths are
+     *           matched, returns the length of the first modified array. If <code>path</code> doesn't
+     *           exist or the value at <code>path</code> is not an array, an error is raised.
+     *     </ul>
+     *     If the index is out of bounds or <code>key</code> doesn't exist, an error is raised.
+     * @example
+     *     <pre>{@code
+     * Json.set(client, "doc", "$", "[[], [\"a\"], [\"a\", \"b\"]]").get();
+     * var newValues = new GlideString[] { gs("\"c\""), gs("{\"key\": \"value\"}"), gs("true"), gs("null"), gs("[\"bar\"]") };
+     * var res = Json.arrinsert(client, gs("doc"), gs("$[*]"), 0, newValues).get();
+     * assert Arrays.equals((Object[]) res, new int[] { 5, 6, 7 }); // New lengths of arrays after insertion
+     * var doc = Json.get(client, "doc").get();
+     * assert doc.equals("[[\"c\", {\"key\": \"value\"}, true, null, [\"bar\"]], [\"c\", {\"key\": \"value\"}, "
+     *     + "true, null, [\"bar\"], \"a\"], [\"c\", {\"key\": \"value\"}, true, null, [\"bar\"], \"a\", \"b\"]]");
+     *
+     * Json.set(client, "doc", "$", "[[], [\"a\"], [\"a\", \"b\"]]").get();
+     * res = Json.arrinsert(client, gs("doc"), gs("."), 0, new GlideString[] { gs("\"c\"") }).get();
+     * assert res == 4 // New length of the root array after insertion
+     * doc = Json.get(client, "doc").get();
+     * assert doc.equals("[\"c\", [], [\"a\"], [\"a\", \"b\"]]");
+     * }</pre>
+     */
+    public static CompletableFuture<Object> arrinsert(
+            @NonNull BaseClient client,
+            @NonNull GlideString key,
+            @NonNull GlideString path,
+            int index,
+            @NonNull GlideString[] values) {
+        return executeCommand(
+                client,
+                new ArgsBuilder()
+                        .add(gs(JSON_ARRINSERT))
+                        .add(key)
+                        .add(path)
+                        .add(Integer.toString(index))
```

Comment:
do we need to convert string to gs?

Reply:
No need, it gets converted automatically.